CHANGES.md

```diff
@@ -1,6 +1,463 @@
+   https://chapel-lang.org/docs/master/builtins/internal/OwnedObject.html#coercions-for-owned))
+* added support for assigning 'nil' to owned and shared classes
+* made the default intent for formals declared with owned/shared type 'const in'
+* made forall loops borrow outer owned and shared class objects by default
+* improved the appearance of shared objects printed via writeln()
+* added support for argument types containing nested generic or query types
+  (see 'Formal Arguments of Partially Generic Type' in the 'Generics' chapter)
+* improved the instantiation of generic arguments when coercions are involved
+* added support for infix bool operators to throw
+* improved isSubtype() and isProperSubtype() to handle cases that can coerce
+* extended isSubtype() and isProperSubtype() to work with generic types
+  (see https://chapel-lang.org/docs/master/builtins/internal/UtilMisc_forDocs.html#UtilMisc_forDocs.isSubtype)
+* improved disambiguation and visibility rules when resolving function calls
+  (see 'Function Visibility in Generic Functions' in the 'Generics' chapter)
+* made 'in' arguments more consistent with variable initialization
+  (see 'Argument Intents' in the language specification's 'Procedures' chapter)
+  (see https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/13.rst)
+* added support for casts from 'c_void_ptr' to 'c_ptr'
+
+Removed Features
+----------------
+* removed implicit 'outer' field on nested types with initializers
+* removed the 'UtilReplicatedVar' module (deprecated in 1.17)
+* disabled non-`ref` forall intents when using recursive parallel iterators
+
+Standard Modules / Library
+--------------------------
+* increased the use of error-handling in standard modules
+* deprecated "out error" patterns in libraries in favor of throwing versions
+* converted some non-fatal checks from using `halt()` to `warning()`
+* deprecated 'Owned(C)' and 'Shared(C)' in favor of 'owned C' and 'shared C'
+  (see TODO)
+* added type query routines for 'owned', 'shared', 'unmanaged', and 'borrowed'
+  (see https://chapel-lang.org/docs/master/modules/standard/Types.html#Types.isOwnedClass)
+* added support for singular value decomposition to the `LinearAlgebra` module
+  (see https://chapel-lang.org/docs/master/modules/packages/LinearAlgebra.html#LinearAlgebra.svd)
+* enabled using `LinearAlgebra` without having `BLAS/LAPACK` installed
+  (see https://chapel-lang.org/docs/master/modules/packages/LinearAlgebra.html#compiling-with-linear-algebra)
+* added configs `blasImpl` and `blasHeader` to `BLAS` to select backend
+  (see https://chapel-lang.org/docs/master/modules/packages/BLAS.html#BLAS.blasImpl)
+* added configs `lapackImpl` and `lapackHeader` to `LAPACK` to select backend
+  (see https://chapel-lang.org/docs/master/modules/packages/LAPACK.html#LAPACK.lapackImpl)
+* deprecated `BLAS` config `isBLAS_MKL` in favor of using `blasImpl`
+* made LinearAlgebra sparse matrices store indices in unsorted order
```

Comment:
Just occurred to me - It would be nice to include the impact of this somewhere, e.g. 

```
* improved the performance of sparse matrix multiplication in `LinearAlgebra`
```